src/imageselector.cpp

```diff
@@ -12,127 +12,172 @@
 #include <algorithm>    // std::shuffle
 #include <random>       // std::default_random_engine
 
-ImageSelector::ImageSelector(std::unique_ptr<PathTraverser>& pathTraverser, char aspectIn):
-  pathTraverser(pathTraverser), aspect(aspectIn)
+ImageSelector::ImageSelector(std::unique_ptr<PathTraverser>& pathTraverser):
+  pathTraverser(pathTraverser)
 {
 }
 
 ImageSelector::~ImageSelector(){}
 
-int ImageSelector::getImageRotation(const std::string& fileName)
+void ImageSelector::setDebugMode(bool debugModeIn) 
+{ 
+  debugMode = debugModeIn;
+}
+
+int ReadExifTag(ExifData* exifData, ExifTag tag, bool shortRead = false)
+{
+  int value = -1;
+  ExifByteOrder byteOrder = exif_data_get_byte_order(exifData);
+  ExifEntry *exifEntry = exif_data_get_entry(exifData, tag);
+
+  if (exifEntry)
+  {
+    if (shortRead)
+    {
+      value = exif_get_short(exifEntry->data, byteOrder);
```

Comment:
can you return the value here so we don't need that much if-else logic?

Reply:
Done!